File: Python Excel Examples/RangesApiDemo/copyRanges.py
import spirecloudexcel
from spirecloudexcel.configuration import Configuration as ExcelConfiguration

appId = "your id"
appKey = "your key"
baseUrl="https://api.e-iceblue.cn"
configuration = ExcelConfiguration(appId, appKey,baseUrl)
api = spirecloudexcel.api.ranges_api.RangesApi(configuration)

name = "CopyRanges.xlsx"
folder = "Ranges"
storage = ""
password = ""
sheetName = "Sheet1"
rangeOperate = spirecloudexcel.models.RangeCopyRequest()
rangeOperate.operate = "copystyle"  # Available values: copystyle, copydata, copyvalue, copyto
PasteOptions=spirecloudexcel.models.PasteOptions()
PasteOptions.only_visible_cells=True
range1 = spirecloudexcel.models.Range()
range1.first_column = 1
range1.first_row = 1
range1.row_count = 10
range1.column_count = 10
range2 = spirecloudexcel.models.Range()
range2.first_column = 1
range2.first_row = 30
range2.row_count = 5
range2.column_count = 8
rangeOperate.source = range1
rangeOperate.target = range2
api.copy_ranges(name, sheetName, rangeOperate, storage = storage, password=password,folder = folder)

File: JavaScript Excel Examples/RangesApiDemo/copyRanges.js
(function (){
    var appId = "your id";
    var appKey = "your key";
    var baseUrl = "https://api.e-iceblue.cn";

    var SpirecloudExcel = require('../../src/index');
    var apiClient = new SpirecloudExcel.ApiClient(appId, appKey,baseUrl);
    var instance = new SpirecloudExcel.RangesApi(apiClient);
    
    var name = "copyRanges.xlsx";
    var sheetName = "Sheet1";
    var opts = { folder: "input" };
    var rangeOperate = new SpirecloudExcel.RangeCopyRequest();
    rangeOperate.operate = "copystyle";  
    rangeOperate.source = new SpirecloudExcel.Range();
    rangeOperate.source.ColumnCount = 10;
    rangeOperate.source.RowCount = 10;
    rangeOperate.source.FirstColumn = 1;
    rangeOperate.source.FirstRow = 1;
    rangeOperate.target = new SpirecloudExcel.Range();
    rangeOperate.target.ColumnCount = 8;
    rangeOperate.target.RowCount = 5;
    rangeOperate.target.FirstColumn = 1;
    rangeOperate.target.FirstRow = 30;
    instance.copyRanges(name, sheetName, rangeOperate, opts, 
      function (error, data, response) {
        if (error) throw error;
    });
})();